wire-ios-utilities/Source/UserDefaults/NativelySupportedUserDefaultsKey.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+/// A protocol which defines
+public protocol NativelySupportedUserDefaultsKey {
```

Comment:
I'm not sure if I understand why this protocol is needed. Is it supposed to be syntax sugar around `UserDefaults`?

Reply:
This protocol, the concrete implementations (currently only `UserDefaults*Date*ValueKey`) and the extension to `UserDefaults` in [UserDefaults+NativelySupportedKey.swift](https://github.com/wireapp/wire-ios/pull/1427/files#diff-f3f3681b271fe518f02ba52c562093272a117ada0a2b6bd6f9d2d654b17fcbac) together allow for having typed UserDefaults methods for reading and writing values.

So one can define UserDefaults keys together with the correct type, like `lastTimeNotificationPermissionHintWasShown` which refers to date values:
```
userDefaults.setValue(
    /*accepting only Date values*/,
    for: .lastTimeNotificationPermissionHintWasShown
)

userDefaults.value(
    for: .lastTimeNotificationPermissionHintWasShown
) // returning optional Date
```

If there is a consensus about removing it, I will get rid of it.

Reply:
Okay let's use it and see how it works